internal/controller/apiextensions/claim/configurator.go

```diff
@@ -86,14 +86,17 @@ func ConfigureComposite(_ context.Context, cm resource.CompositeClaim, cp resour
 		return errors.New(errUnsupportedClaimSpec)
 	}
 
-	// Delete base claim fields when configuring composite spec
-	baseClaimSpec := xcrd.CompositeResourceClaimSpecProps()
-
-	// keep any necessary fields between claim and composite
-	for _, field := range xcrd.KeepClaimSpecProps {
-		delete(baseClaimSpec, field)
-	}
-	claimSpecFilter := xcrd.GetPropFields(baseClaimSpec)
+	// We want to propagate the claim's spec to the composite's spec, but
+	// first we must filter out any well-known fields that are unique to
+	// claims. We do this by:
+	// 1. Grabbing a map whose keys represent all well-known claim fields.
+	// 2. Deleting any well-known fields that we want to propagate.
+	// 3. Using the resulting map keys to filter the claim's spec.
+	wellKnownClaimFields := xcrd.CompositeResourceClaimSpecProps()
+	for _, field := range xcrd.PropagateClaimSpecProps {
+		delete(wellKnownClaimFields, field)
+	}
+	claimSpecFilter := xcrd.GetPropFields(wellKnownClaimFields)
```

Comment:
There's no logic change here - I just got stuck on this code for a bit and wanted to try make it a bit more readable.